### DISCUSSION TITLE
ILI9488 Display support

### DISCUSSION THREAD

**Discussion by summongod** [arendst/Tasmota#10630] [Q&A] [answered]
Hi, I'm trying to get an ILI9488 working with an ESP32. I initially thought this would be similar to the ILI9341 but on looking through the code I can see that different implementation strategies have been used. (9341 uses GFX whereas 9488 uses it's own renderer).

My question is, are people using this screen at the moment and if so what level of functionality should I expect wrt the 9341? I currently cannot get anything working and will abandon trying if this display is a dead end.

**Comment by gemu2015** [comment]
you have to jumper the display to 3 wire SPI mode, then connect SPI_MOSI,  SPI_CLK, ILI9488_CS and Backlight. (so no SPI_DC)
the renderer class is just an intermediate between the displays and the GFX driver supporting additional Fonts.
in user_override define

#define USE_DISPLAY
#define USE_SPI
#define USE_DISPLAY_ILI9488

and if you want to use Touch
#define USE_FT5206

**Reply by summongod** [reply]
I'm having no luck here :-( Could I ask you what display you are using please? Mine is 
![image](https://user-images.githubusercontent.com/54992222/105064448-bdaefc80-5a74-11eb-9d81-25399c86452a.png)

I have it working using arduino but it uses many more GPIOs and the clks &  mosi/din are connected.
Can it be I'm doing something strange or perhaps its a different screen?

**Comment by gemu2015** [comment]
it seems yours can not be configured to 3 wire spi

this is mine with capacitve touch:

https://www.buydisplay.com/lcd-3-5-inch-320x480-tft-display-module-optl-touch-screen-w-breakout-board

**Reply by summongod** [reply]
My one has a resistive touch I think. I can't find your screen for purchase in the UK so I think I might roll back to the ILI9341.
I thought that 9.2 release display binary would support both the 9341 and 9488 as standard but apparently not. Dev 9.2.0.3(display) has ILI9488 CS and ILI9341 CS selectable in parameters.

**Reply by damondins** [reply]
> it seems yours can not be configured to 3 wire spi
> 
> this is mine with capacitve touch:
> 
> https://www.buydisplay.com/lcd-3-5-inch-320x480-tft-display-module-optl-touch-screen-w-breakout-board

@gemu2015 I bought the display that you have and I'm having a hard time understanding what to do to configure it as 3 wire spi. I also dont know which pins to use. I'm connecting this to Tasmota with a wemos D1 mini.

Thanks in advance,
Damon

**Reply by gemu2015** [reply]
normally you can buy it preconfigured at buy display.
otherwise you have to check the data sheet which jumpers to change. there are dedicated jumpers for SPI mode.
in Tasmota GPIO config you must select and connect to appropriate pins on ILI9488 board
SPI_MOSI = GPIO 13
SPI_MISO = GPIO 12
SPI_CLK = GPIO 14
CS_ILI9488 = GPIO 15
Backlight = any other

in config_user_override
#define USE_SPI
#define USE_DISPLAY
#define USE_DISPLAY_ILI9488
#define SHOW_SPLASH

if you want to use touch, define i2c on 2 remaining pins and connect to ILI9488 board i2c

#define USE_I2C
#define USE_FT5206
#define USE_TOUCH_BUTTONS

if you want sd card connect SPI also to SD card pins on ILI9488
choose a free GPIO and connect to ILI9488 SDcard CS
and set this pin in Tasmota GPIO to CS_SDCARD

and 
#define USE_UFILE
#define USE_UFILESYS
#define USE_SDCARD

**Reply by gemu2015** [reply]
ILI9488 assembly on ESP8266

23 LCD_/CS = ILI9488 CS GPIO15
24 SCL = SPI_SCK GPIO14
27 LCD_SDI = SPI_MOSI GPIO13
28 LCD_SDO = SPI_MISO GPIO12
29 BL_ON/OFF = Backlight (any pin)
30 CTP_SCL = I2C_SCL (any pin)
31 CTP_SDA = I2C_SDA (any pin)
32 SDO = SPI_MISO GPIO12
33 SCL = SPI_SCK GPIO14
34 SDI = SPI_MOSI GPIO13
35 SD_/CS = SDCARD_CS (any pin)


jumpers soldered for 3 wire SPI
J1,J3,J5 Short and J2,J4,J6 Open.
R1-R10=0R, R17=0R,R19=0R,R21-R28=0R and R18,R20 not soldering.

**Comment by gemu2015** [comment, marked as answer]
it currently only supports 3 wire spi on the ili9488 and capacitive touch. my display is from buydisplay and arrived in about 10 days in Germany. there is currently no resistive touch display support on Tasmota. (i would strongly recommend capacitive  touch)

**Comment by summongod** [comment]
Thanks for responding. I think I will continue with the 9341 and use the 9488 on a different project.

**Comment by damondins** [comment]
That's what I needed. I'll try it and let you know.
